src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/Serialization/SymbolSpecification.cs

```diff
@@ -65,14 +70,21 @@ public SymbolSpecification(Guid id, string symbolSpecName, List<SymbolKindOrType
             RequiredCustomTagList = customTagList;
         }
 
+        public SymbolSpecification(string symbolSpecName, SymbolKindOrTypeKind kind, ModifierKindEnum modifierList, List<AccessibilityKind> accessibilityList, string guidString)
+            : this(Guid.Parse(guidString), symbolSpecName, SpecializedCollections.SingletonEnumerable(kind), accessibilityList, modifierList, SpecializedCollections.EmptyEnumerable<string>())
+        {
+
+        }
+
         internal bool AppliesTo(ISymbol symbol, ISymbolCategorizationService categorizationService)
         {
             if (ApplicableSymbolKindList.Any() && !ApplicableSymbolKindList.Any(k => k.AppliesTo(symbol)))
             {
                 return false;
             }
-
-            if (!RequiredModifierList.All(m => m.MatchesSymbol(symbol)))
+            
+            // Modifiers must match exactly
+            if (RequiredModifierList != ModifierKind.GetModifiers(symbol))
             {
```

Comment:
I was about to suggest SequenceEqual, and then I realize RequiredModifierList is not a list of things but an enum value. That's not clear from reading this code, perhaps a better name would help?